src/Parser.php

```diff
@@ -286,50 +288,20 @@ private function tokenTypes()
 
     private function getMatchingJsonPointerPath(): array
     {
-        $matchingPointer = key($this->paths);
-
-        if (count($this->paths) === 1) {
-            $this->matchedJsonPointer = $matchingPointer;
-
-            return $this->paths[$matchingPointer];
-        }
-
-        $currentPathLength = count($this->currentPath);
-        $matchLength = -1;
-
-        foreach ($this->paths as $jsonPointer => $path) {
-            $matchingReferenceTokens = [];
-
-            foreach ($path as $i => $referenceToken) {
-                if (
-                    ! isset($this->currentPath[$i])
-                    || (
-                        $this->currentPath[$i] !== $referenceToken
-                        && ValidJsonPointers::wildcardify($this->currentPath[$i]) !== $referenceToken
-                    )
-                ) {
-                    continue;
+        $matchingPointerByIndex = [];
```

Comment:
simplifying and making the logic to compute the matching JSON pointer more accurate